src/encoded/types/individual.py

```diff
@@ -118,3 +137,114 @@ class IndividualZebrafish(Individual):
     item_type = 'individual_zebrafish'
     schema = load_schema('encoded:schemas/individual_zebrafish.json')
     embedded_list = Individual.embedded_list + ['zebrafish_vendor.name']
+
+
+# validator for individual relations - same species
+def validate_individual_relations(context, request):
+    '''Validates that individual relations are within the same species,
+    limited to two relations per individual (max 1 paternal and 1 maternal),
+    not self relations, and unique (no duplicate relations).
+    '''
+    data = request.json
+    related_individuals = data.get('individual_relation')  # a list of dicts
+    if related_individuals is None:
+        return
+    any_failures = False
+    # check if the individual info is in request (POST) or in context (PATCH)
+    try:
```

Comment:
I was having issues deploying local and loading the page when I pulled your branch. The following suggestion (replacing the whole try/except block) fixed the issue for me, though rather than just accepting it you might want to try yourself and see if Andy has any input since he might have more experience with this part of the code than me. 

```suggestion
    individual_uuid = data.get('uuid')
    organism_id = data.get('organism')
    # try:
    #     individual_uuid = data['uuid']
    #     organism_id = data.get('organism')
    # except KeyError:
    #     individual_uuid = str(context.uuid)
    #     individual = get_item_if_you_can(request, individual_uuid, 'individuals')
    #     organism_id = individual.get('organism')
```

Reply:
Further note: organism is a required property for individual, so it should always be present and I don't think it would ever make sense to patch a different organism. uuid should also always be in the post, unless I'm misunderstanding the process.

Reply:
I could deploy local without issues, so I am not sure what the issue might be.

The whole try/except block is necessary because otherwise the patch test fails. When you patch just the individual_relations (as in the test), the information about the individual being patched (e.g. its uuid) is not in the variable request, but in context. This block accounts for that.

Reply:
Alright, I'll take your word for it then and look into why I can't deploy.